test/coverage/datamodel/UUriValidatorTest.cpp

```diff
@@ -23,16 +27,296 @@ class TestFixture : public testing::Test {
 
 	// Run once per execution of the test application.
 	// Used for setup of all tests. Has access to this instance.
-	TestFixture() = default;
-	~TestFixture() = default;
+	TestUUriValidator() = default;
+	~TestUUriValidator() = default;
 
 	// Run once per execution of the test application.
 	// Used only for global setup outside of tests.
 	static void SetUpTestSuite() {}
 	static void TearDownTestSuite() {}
 };
 
-// TODO replace
-TEST_F(TestFixture, SomeTestName) {}
+TEST_F(TestUUriValidator, Valid) {
+	uprotocol::v1::UUri uuri;
+	uuri.set_authority_name(AUTHORITY_NAME);
+	uuri.set_ue_id(0x00010001);
+	uuri.set_ue_version_major(1);
+	uuri.set_resource_id(1);
+
+	uuri.set_authority_name("");
+	auto res = isValid(uuri);
+	EXPECT_FALSE(std::get<bool>(res));
+	EXPECT_TRUE(std::get<std::optional<Reason>>(res) == Reason::EMPTY);
+	uuri.set_authority_name(AUTHORITY_NAME);
```

Comment:
I would recommend structuring a test like this such that a new uuri is created for each configuration that is checked.  The pattern I typically use looks like this:
```suggestion
    auto getUuri = []() {
	    uprotocol::v1::UUri uuri;
	    uuri.set_authority_name(AUTHORITY_NAME);
	    uuri.set_ue_id(0x00010001);
	    uuri.set_ue_version_major(1);
	    uuri.set_resource_id(1);
	    return uuri;
	};

	{
	    auto uuri = getUuri();
	    auto [valid, maybeReason] = isValid(uuri);
	    EXPECT_TRUE(valid);
	    EXPECT_FALSE(maybeReason);
	}

    {
	    auto uuri = getUuri();
	    uuri.set_authority_name("");
	    auto [valid, maybeReason] = isValid(uuri);
	    EXPECT_FALSE(valid);
	    EXPECT_TRUE(maybeReason);
	    EXPECT_EQ(*maybeReason, Reason::EMPTY);
	}
```

Reply:
Done.  I do have to admit, I was originally against this change and felt it was unnecessary, however after doing the change, I can honestly say I like it a LOT BETTER than before.

Reply:
Thanks for making the change. It's a lot easier to read what the test is doing now.